src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java

```diff
@@ -534,6 +535,306 @@ public void testRequiredIssuerWithIncorrectAudience() {
+
+        String jwsToken1 = Jwts.builder().setSubject("Leonard McCoy").signWith(priv1, SignatureAlgorithm.RS256).compact();
+
+        String jwsToken2 = Jwts.builder().setSubject("Stephen Crawford").signWith(priv2, SignatureAlgorithm.ES512).compact();
+
+        Settings settings = Settings.builder()
+            .put(
+                "signing_key",
+                "-----BEGIN PUBLIC KEY-----\n"
+                    + BaseEncoding.base64().encode(pub1.getEncoded())
+                    + "-----END PUBLIC KEY-----,"
+                    + BaseEncoding.base64().encode(pub2.getEncoded())
+            )
+            .build();
+
+        HTTPJwtAuthenticator jwtAuth = new HTTPJwtAuthenticator(settings, null);
+        Map<String, String> headers1 = new HashMap<String, String>();
+        headers1.put("Authorization", "Bearer " + jwsToken1);
+
+        AuthCredentials creds1 = jwtAuth.extractCredentials(
+            new FakeRestRequest(headers1, new HashMap<String, String>()).asSecurityRequest(),
+            null
+        );
+
+        Assert.assertNotNull(creds1);
+        assertThat(creds1.getUsername(), is("Leonard McCoy"));
+        assertThat(creds1.getBackendRoles().size(), is(0));
+
+        Map<String, String> headers2 = new HashMap<String, String>();
+        headers2.put("Authorization", "Bearer " + jwsToken2);
+        AuthCredentials creds2 = jwtAuth.extractCredentials(
+            new FakeRestRequest(headers2, new HashMap<String, String>()).asSecurityRequest(),
+            null
+        );
+
+        Assert.assertNotNull(creds2);
+        assertThat(creds2.getUsername(), is("Stephen Crawford"));
+        assertThat(creds2.getBackendRoles().size(), is(0));
+    }
+
+    @Test
+    public void testManyMultipleSigningKeysMixedAlgsParseSuccessfully() throws NoSuchAlgorithmException {
+
+        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+        keyGen.initialize(2048);
+        KeyPair pair1 = keyGen.generateKeyPair();
+        PrivateKey priv1 = pair1.getPrivate();
+        PublicKey pub1 = pair1.getPublic();
+
+        KeyPairGenerator keyGen2 = KeyPairGenerator.getInstance("EC");
+        keyGen2.initialize(521);
+        KeyPair pair = keyGen2.generateKeyPair();
+        PrivateKey priv2 = pair.getPrivate();
+        PublicKey pub2 = pair.getPublic();
+
+        KeyPairGenerator keyGen3 = KeyPairGenerator.getInstance("RSA");
+        keyGen3.initialize(2048);
+        KeyPair pair3 = keyGen3.generateKeyPair();
+        PrivateKey priv3 = pair3.getPrivate();
+        PublicKey pub3 = pair3.getPublic();
+
+        KeyPairGenerator keyGen4 = KeyPairGenerator.getInstance("EC");
+        keyGen4.initialize(521);
+        KeyPair pair4 = keyGen4.generateKeyPair();
+        PrivateKey priv4 = pair4.getPrivate();
+        PublicKey pub4 = pair4.getPublic();
+
+        String jwsToken1 = Jwts.builder().setSubject("Stephen Crawford").signWith(priv1, SignatureAlgorithm.RS256).compact();
```

Comment:
nit: I know it may work for the test, but usernames should not contain whitespace

Reply:
All the tests do this. I think it is fine.